nuget/chocolatey/tools/chocolateysetup.psm1

```diff
@@ -39,11 +39,17 @@ param(
 
   Install-DotNet4IfMissing
 
+  if ($env:ChocolateyEnvironmentAllowGlobalConfirmation -eq 'true') {
+    $allowGlobalConfirmation = '-y'
+  } else {
+    $allowGlobalConfirmation = $null
+  }
+
   $chocoNew = Join-Path $thisScriptFolder 'chocolateyInstall\choco.exe'
   if ($debugMode) {
-    & $chocoNew unpackself -fdv
+    & $chocoNew unpackself -fdv $allowGlobalConfirmation

```

Comment:
This makes me think we should set $debugMode as `'-dv'` and only have one command there instead of two. But also that we should always use `-y`. If the script stops for any reason, it is breaking Chocolatey conventions.